test.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * A tool for executing all of the `acceptance-tests` tests and formatting
+ * their results into grokkable output.
+ */
+
+var locations = require( './locations.json' );
+var util = require( 'util' );
+var querystring = require( 'querystring' );
+var supertest = require( 'supertest' );
+var colors = require( 'colors' );
+var commander = require( 'commander' );
+var requireDir = require( 'require-dir' );
+var peliasConfig = require( 'pelias-config' ).generate();
+var terminalOutputGenerator = require( './output_generators/terminal.js' );
+
+/**
+ * Return a boolean indicating whether `actual` has all the key value pairs
+ * contained in `expected`.
+ */
+function equalProperties( expected, actual ){
+  for( var prop in expected ){
+    if( actual[ prop ] !== expected[ prop ] ){
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Given a test-case, the API results for the input it specifies, and a
+ * priority-threshold to find the results in, return an object indicating the
+ * status of this test (whether it passed, failed, is a placeholder, etc.)
+ */
+function evalTest( priorityThresh, testCase, apiResults ){
+  var expected;
+  if( typeof testCase.out === 'string' ){
+    if( testCase.out in locations ){
+      expected = locations[ testCase.out ];
+    }
+    else {
+      return {
+        result: 'placeholder',
+        msg: 'Placeholder test, no `out` object matches in `locations.json`.'
+      }
+    }
+  }
+  else if( !( 'out' in testCase ) || testCase.out === null ){
+    return {
+      result: 'placeholder',
+      msg: 'Placeholder test, no `out` specified.'
+    };
+  }
+  else {
+    expected = testCase.out;
+  }
+
+  for( var ind = 0; ind < apiResults.length; ind++ ){
+    var result = apiResults[ ind ];
+    if( equalProperties( expected, result.properties ) ){
+      var success = ( ind + 1 ) <= priorityThresh;
+      return ( success ) ?
+        { result: 'pass' } :
+        {
+          result: 'fail',
+          msg: util.format( 'Result found, but not in top %s.', priorityThresh )
+        }
+    }
+  }
+
+  return {
+    result: 'fail',
+    msg: 'No result found.'
+  }
+}
+
+/**
+ * Execute all the tests in a test-suite file with `evalTest()`, and pass an
+ * object containing the results to `cb()`. `apiUrl` contains the URL of the
+ * Pelias API to query.
+ */
+function execTestSuite( apiUrl, testSuite, cb ){
+  var testResults = {
+    stats: {
+      pass: 0,
+      fail: 0,
+      placeholder: 0,
+      timeTaken: null,
+      name: testSuite.name
+    },
+    results: []
+  };
+
+  var startTime = new Date().getTime();
```

Comment:
It's not really a benchmark and added as somewhat of stat-sugar, so I'd opt for the simplicity of `new Date()`. Plus, it's not totally accurate because these tests get executed in parallel (meaning `startTime` gets initialized immediately, while the API processes results more or less sequentially and the time per suite gets compounded with the time the preceding suites took to execute), but that's another problem. I can change over to `microtime`, though I'm not sure if it matters here.

Should tests be made sequential?